apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/CreationTimeFilter.java

```diff
@@ -0,0 +1,21 @@
+package com.linkedin.openhouse.jobs.util;
+
+import java.util.concurrent.TimeUnit;
+import lombok.AllArgsConstructor;
+
+/** Filter class to collect table with creation time less than currentTime - cutoffHour */
+@AllArgsConstructor
+public class CreationTimeFilter implements TableFilter {
+
+  private int cutoffHour;
+
+  public static CreationTimeFilter of(int timeInHour) {
+    return new CreationTimeFilter(timeInHour);
+  }
+
+  @Override
+  public boolean apply(TableMetadata metadata) {
+    return metadata.getCreationTime()
+        < System.currentTimeMillis() - TimeUnit.HOURS.toMillis(cutoffHour);
```

Comment:
Looks like cutoffHour is not initialized?

Reply:
Class has AllArgsConstructor, so cutOffHour will be initialized with that.